syntax = "proto3";
package swap_internal;

service SwapManager {
    rpc AddAddressFilters (AddAddressFiltersRequest) returns (AddAddressFiltersReply) {}
    rpc GetInfo (GetInfoRequest) returns (GetInfoReply) {}
    rpc GetSwap (GetSwapRequest) returns (GetSwapReply) {}
    rpc Stop (StopRequest) returns (StopReply) {}
}

message AddAddressFiltersRequest {
    repeated string addresses = 1;
}

message AddAddressFiltersReply {}

message GetInfoRequest {}
message GetInfoReply {
    uint64 block_height = 1;
    string network = 2;
}

message GetSwapRequest {
    optional string address = 1;
    optional string payment_request = 2;
    optional bytes payment_hash = 3;
}
message GetSwapReply {
    string address = 1;
    repeated SwapOutput outputs = 2;
}
message SwapOutput {
    string outpoint = 1;
    optional uint64 confirmation_height = 2;
}

message StopRequest {}
message StopReply {}